### ISSUE TITLE
SVG tags' shapes lose their closing tags

### ISSUE THREAD

**Issue by brian-c** [cheeriojs/cheerio#521]
```js
var cheerio = require('cheerio');
var $ = cheerio.load('<svg><symbol><circle></circle></symbol><g><rect></rect></g><path></path></svg>');
console.log($.html());
//=> '<svg><symbol><circle></symbol><g><rect></g><path></svg>'
```

Using 0.17.0, `<circle>`, `<rect>`, and `<path>` lose their closing tags.

**Comment by silverwind** [comment]
This happened to me to with this data in regular (non-xml) mode. Cheerio removed the close tags and the `rect` and `path` nodes ended up being interpreted as nested:
````
<svg>
    <rect/>
    <path/>
    <rect/>
    <path/>
</svg>
````
Adding explicit close tag didn't help either, and my only solution (except xml mode, but that's not an option for me) was to wrap the `rect` and `path` nodes inside `<g>` tags.

**Comment by fb55** [comment]
Will be fixed with #673.

**Comment by silverwind** [comment]
:+1:

**Comment by bezoerb** [comment]
:+1

**Comment by kaemu** [comment]
Thank you for the fix #673. It's ok for most of the tags (path, line, polyline…) excepted <polygon> which is not closed, and break the svg.
